src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -367,11 +271,30 @@ def visit_Call(self, node: foast.Call, **kwargs) -> foast.Call:
                 else:
                     new_dims.extend(target_dims)
             new_type = ct.FieldType(dims=new_dims, dtype=new_func.type.dtype)
-            return foast.Call(func=new_func, args=new_args, location=node.location, type=new_type)
+            return foast.Call(
+                func=new_func, args=new_args, kwargs={}, location=node.location, type=new_type
+            )
         elif isinstance(new_func.type, ct.FunctionType):
+            new_args = self.visit(node.args, **kwargs)
+            new_kwargs = {
+                keyword: self.visit(arg, **kwargs) for keyword, arg in node.kwargs.items()
+            }
+            try:
```

Comment:
It does look like a bit of an oversight that signatures are checked only now, but keep in mind that there didn't use to be any builtin functions not too long ago...